packages/framework-core/src/services/event-store.ts

```diff
@@ -49,39 +50,58 @@ export class EventStore {
     entityName: string,
     entityID: UUID,
     pendingEnvelopes: Array<EventEnvelope>
-  ): Promise<EventEnvelope | null> {
+  ): Promise<Array<EventEnvelope>> {
     const logger = getLogger(this.config, 'EventStore#calculateAndStoreEntitySnapshot')
     logger.debug('Processing events: ', pendingEnvelopes)
     logger.debug(`Fetching snapshot for entity ${entityName} with ID ${entityID}`)
     const latestSnapshotEnvelope = await this.loadLatestSnapshot(entityName, entityID)
+    const snapshotsPerNewEntityId: Record<string, EventEnvelope> = {}
+    if (latestSnapshotEnvelope) {
+      snapshotsPerNewEntityId[latestSnapshotEnvelope.entityID.toString()] = latestSnapshotEnvelope
+    }
 
     logger.debug(
       `[EventStore#calculateAndStoreEntitySnapshot] Looking for the reducer for entity ${entityName} with ID ${entityID}`
     )
-    let newEntitySnapshot = latestSnapshotEnvelope
     for (const pendingEvent of pendingEnvelopes) {
-      newEntitySnapshot = await this.entityReducer(newEntitySnapshot, pendingEvent)
+      logger.debug(`Reducing snapshot for entity ${entityName} with ID ${pendingEvent.entityID.toString()}`)
```

Comment:
Wouldn't this message always say the same for all the events? 
If so, I think we can delete it, specially because the `this.entityReducer` method already has its own logs for the same purpose.

Reply:
Right!